flexmeasures/data/models/forecasting/pipelines/base.py

```diff
@@ -404,6 +408,10 @@ def _slice_closed(
                         seconds=index_offset * target_sensor_resolution.total_seconds()
                     )
                     belief_time = first_belief_time + delta
+                    if pd.isna(self.save_belief_time):
+                        save_belief_time = belief_time
+                    else:
+                        save_belief_time = self.save_belief_time + delta
```

Comment:
Please add an inline comment explaining this. This deals with updating the belief time for the next cycle, right?